src/test/test_addr.c

```diff
@@ -1671,8 +1671,53 @@ test_addr_octal(void *arg)
   ;
 }
 
+#define TEST_ADDR_VALIDITY(a, lis, rv) STMT_BEGIN \
+    tt_int_op(tor_addr_is_valid(a, lis), OP_EQ, rv); \
+  STMT_END;
+
+static void
+test_addr_is_valid(void *arg)
+{
+  (void)arg;
+  tor_addr_t *test_addr = tor_malloc(sizeof(tor_addr_t));
+  /* Tests for IPv4 addresses. */
+  test_addr->family = AF_INET;
+
+  /* Obtain 0.0.0.0 . */
+  tor_inet_pton(test_addr->family, "0.0.0.0",
+                                    &test_addr->addr.in_addr);
+  TEST_ADDR_VALIDITY(test_addr, 0, 0);
+  TEST_ADDR_VALIDITY(test_addr, 1, 1);
+
+  /* Obtain 123.90.1.1. */
+  tor_inet_pton(test_addr->family, "123.90.1.1",
+                                    &test_addr->addr.in_addr);
+  TEST_ADDR_VALIDITY(test_addr, 0, 1);
+  TEST_ADDR_VALIDITY(test_addr, 1, 1);
+
+  tor_free(test_addr);
+
+  /* Tests for IPv6 addresses. */
+  test_addr = tor_malloc(sizeof(tor_addr_t));
+  test_addr->family = AF_INET6;
+
+  /* Obtain 2000::1a00::1000:fc098 . */
```

Comment:
This test is hard to read, because the order is:
IPv4 null
IPv4 not null
IPv6 not null
IPv6 null

Please use a consistent order.